src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerMaintainLinkCase.java

```diff
@@ -1,83 +1,110 @@
+                    SUPER_USER,
+                    ST_CIC_JUDGE))
+                .page("maintainCaseLink")
+                .pageLabel("Maintain Case Link")
+                .optional(CaseData::getCaseLinks, ALWAYS_HIDE, null, true)
+                .optional(CaseData::getLinkedCasesComponentLauncher,
+                    null, null, null, null, "#ARGUMENT(UPDATE,LinkedCases)");
         }
     }
 
-    private void doConfigure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
-        PageBuilder pageBuilder = new PageBuilder(configBuilder
-            .event(CASEWORKER_MAINTAIN_LINK_CASE)
-            .forStates(Submitted, CaseManagement, AwaitingHearing, AwaitingOutcome)
-            .name("Links: Maintain Link case")
-            .showSummary()
-            .description("Links: Maintain Link case")
-            .aboutToSubmitCallback(this::aboutToSubmit)
-            .submittedCallback(this::submitted)
-            .grant(CREATE_READ_UPDATE,
-                ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN,
-                ST_CIC_HEARING_CENTRE_TEAM_LEADER));
-
-        linkCaseSelectCase.addTo(pageBuilder);
-    }
-
-    @SneakyThrows
-    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
-                                                                       CaseDetails<CaseData, State> beforeDetails) {
-        log.info("Caseworker link the case callback invoked for Case Id: {}", details.getId());
+    public SubmittedCallbackResponse linkUpdated(CaseDetails<CaseData, State> details,
+                                                 CaseDetails<CaseData, State> beforeDetails) {
         var data = details.getData();
-
-        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .data(data)
-            .state(details.getState())
+        String claimNumber = data.getHyphenatedCaseRef();
+        try {
+            unLinkedCaseNotification(claimNumber, data);
+        } catch (Exception notificationException) {
+            log.error("Case Link notification failed with exception : {}", notificationException.getMessage());
+            return SubmittedCallbackResponse.builder()
+                .confirmationHeader(format("# Case Link updated %n"))
+                .build();
+        }
+        return SubmittedCallbackResponse.builder()
+            .confirmationHeader(format("# Case Link updated %n"))
             .build();
     }
 
-    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
-                                               CaseDetails<CaseData, State> beforeDetails) {
-        return SubmittedCallbackResponse.builder()
-            .confirmationHeader("Case Updated")
-            .build();
+
+    private void unLinkedCaseNotification(String caseNumber, CaseData data) {
+        CicCase cicCase = data.getCicCase();
+
+        if (!cicCase.getSubjectCIC().isEmpty()) {
```

Comment:
Can include null check for all the three conditions.

Reply:
Added null check for the conditions.